[OpenShift Operator] Document securityContext for log/apm via socket (#19525)
DataDog/documentation: content/en/containers/kubernetes/distributions.md

GKE Autopilot requires some configuration, shown below.

> Datadog recommends that you specify resource limits for the Agent container. Autopilot sets a relatively low default limit (50m CPU, 100Mi memory) that may rapidly lead the Agent container to OOMKill depending on your environment. If applicable, also specify resource limits for the Trace Agent and Process Agent containers. Additionally, you may wish to create a priority class for the Agent to ensure it is scheduled.

Reviewer: ```suggestion
Datadog recommends that you specify resource limits for the Agent container. Autopilot sets a relatively low default limit (50m CPU, 100Mi memory) that may lead the Agent container to quickly OOMKill depending on your environment. If applicable, also specify resource limits for the Trace Agent and Process Agent containers. Additionally, you may wish to create a priority class for the Agent to ensure it is scheduled.
```

i see you ran into the linter suggesting you don't use the word "quickly"! don't worry, this is intended to catch marketing-type copy ("this product can quickly solve your problem" etc.)—the situation here is a bit different, as the value of "quickly" here is neutral (or possibly negative). i'm putting "quickly" back

          effect: NoSchedule
```

> **Note**: The nodeAgent Security Context override is necessary for Log Collection and APM - Trace Collection with the `/var/run/datadog/apm/apm.socket` socket. It can be omitted if these features are not enabled.

Reviewer: ```suggestion
**Note**: The nodeAgent Security Context override is necessary for Log Collection and APM Trace Collection with the `/var/run/datadog/apm/apm.socket` socket. If these features are not enabled, you can omit this override.
```
